src/frontends/onnx/frontend/src/input_model.cpp

```diff
@@ -124,9 +124,20 @@ void InputModelONNX::free_name_for_tensor(const std::string&) {
 }
 
 void InputModelONNX::set_partial_shape(Place::Ptr place, const ngraph::PartialShape& shape) {
-    std::map<std::string, ngraph::PartialShape> m;
-    m[place->get_names()[0]] = shape;
-    m_editor->set_input_shapes(m);
+    std::string input_name;  // name of the model input which should be reshaped
+    const auto input_edge = std::dynamic_pointer_cast<PlaceInputEdgeONNX>(place);
+    if (input_edge) {
```

Comment:
It is probably also a topic for separate PR, but probably we can call `get_target_tensor` on the output tensor and handle OutputEdge place

Reply:
I'd like to discuss it a bit more and implement it in a separate PR with proper testing.